crates/wash-cli/src/common/registry_cmd.rs

```diff
@@ -61,7 +101,24 @@ pub async fn registry_pull(
 }
 
 pub async fn registry_ping(cmd: RegistryPingCommand) -> Result<CommandOutput> {
-    let image: Reference = cmd.url.parse()?;
+    let artifact_url = match cmd.url {
```

Comment:
It looks like we have this same logic used 3 times. Can we move it to a function?